### PR TITLE
Feature/simple facets

### PR DESCRIPTION
**Pull request by djtfmartin**
Small enhancement to add facets to list search. Deployed on test server.

Screen Shot 2019-10-08 at 8 12 47 pm

### CODE REVIEW

File: grails-app/controllers/au/org/ala/specieslist/PublicController.groovy [hunk 1]
```diff
@@ -22,30 +22,20 @@ class PublicController {
     }
 
     def speciesLists(){
-        if (params.isSDS) {
-            // work around for SDS sub-list
-            redirect(action:'sdsLists')
-            return
-        }
-        params.max = Math.min(params.max ? params.int('max') : 25, 100)
+        params.max = Math.min(params.max ? params.int('max') : 25, 1000)
```
Comment (nickdos) [inline grails-app/controllers/au/org/ala/specieslist/PublicController.groovy]: I see that the check for SDS lists as well as _not_ showing private lists has been moved into the service, which is a good idea. I'm wondering if a few integration/functional tests should be added that test those conditions, to ensure they are handled correctly and don't break when code is changed?

Comment (nickdos) [reply grails-app/controllers/au/org/ala/specieslist/PublicController.groovy]: I'm not 100% sure the `isSDS` case is handled in the service... hard to tell just reading code but I'm assuming its now a facet that is just like the other fields...

File: grails-app/taglib/au/org/ala/specieslist/SpeciesListTagLib.groovy [hunk 2]
```diff
@@ -58,4 +58,37 @@ class SpeciesListTagLib {
                 .replaceAll("\\(", "").replaceAll("\\)", "")
                 .toLowerCase()
     }
+
+
+    def selectedFacetLink = { attrs ->
```
Comment (nickdos) [inline grails-app/taglib/au/org/ala/specieslist/SpeciesListTagLib.groovy]: taglib methods are quite easy to unit test... might be worth adding some for these

File: grails-app/views/_speciesList.gsp [hunk 3]
```diff
@@ -67,114 +67,159 @@
...
-        </tbody>
-    </table>
+            </thead>
+            <tbody>
+            <g:each in="${lists}" var="list" status="i">
+                <tr class="${(i % 2) == 0 ? 'odd' : 'even'}">
+                    <td>
+                        <a href="${request.contextPath}/speciesListItem/list/${list.dataResourceUid}">${fieldValue(bean: list, field: "listName")}
+                        </a>
+    %{--                    <br/>--}%
```
Comment (nickdos) [inline grails-app/views/_speciesList.gsp]: commented-out code...

File: grails-app/views/_speciesList.gsp [hunk 4]
```diff
@@ -67,114 +67,159 @@
...
-                    <td>
-                        <a href="#"
-                           onclick="fancyConfirm('Are you sure that you would like to rematch ${list.listName.encodeAsHTML()}', ${list.id}, 'rematch');
-                           return false;" id="rematch_${list.id}" class="buttonDiv">Rematch</a>
-                    </td>
-                    <td>
-                        <a href="${request.contextPath}/speciesList/upload/${list.dataResourceUid}"
-                           class="buttonDiv">Reload</a>
-                    </td>
+%{--                <g:sortableColumn property="isAuthoritative" params="${[q:params.q]}"--}%
```
Comment (nickdos) [inline grails-app/views/_speciesList.gsp]: remove these?

File: grails-app/conf/application.groovy [hunk 5]
```diff
@@ -185,49 +185,49 @@ grails {
...
-            url = "jdbc:h2:mem:testDb;MVCC=TRUE;LOCK_TIMEOUT=10000;MODE=MYSQL;DB_CLOSE_ON_EXIT=FALSE;"
-        }
-    }
-    production {
-        println "setting up production datasource"
-        dbCreate = "update"
-        // must be set via external config
-    }
-}
+//
```
Comment (nickdos) [inline grails-app/conf/application.groovy]: delete this section?

Comment (djtfmartin) [reply grails-app/conf/application.groovy]: done

File: grails-app/views/_speciesList.gsp (hunk 4 above)
Comment (djtfmartin) [reply grails-app/views/_speciesList.gsp]: done

File: grails-app/views/_speciesList.gsp (hunk 3 above)
Comment (djtfmartin) [reply grails-app/views/_speciesList.gsp]: removed

File: grails-app/taglib/au/org/ala/specieslist/SpeciesListTagLib.groovy (hunk 2 above)
Comment (djtfmartin) [reply grails-app/taglib/au/org/ala/specieslist/SpeciesListTagLib.groovy]: good feedback. ive added taglib tests.

File: grails-app/controllers/au/org/ala/specieslist/PublicController.groovy (hunk 1 above)
Comment (djtfmartin) [reply grails-app/controllers/au/org/ala/specieslist/PublicController.groovy]: ive put back the isSDS=true check, but removed the redirect to public/sdsLists - that URL now redirects to public/speciesLists?isSDS=eq:true. Hopefully this is more consistent for users - one UI with filters on the left hand side.

**Pull request merged**